ehsm_kms_service/secret_manager_apis.js

```diff
@@ -0,0 +1,279 @@
+            }
+        }
+    }
+    return val
+}
+
+function toBase64(val) {
+    if (typeof val == 'string') {
+        return base64_encode(val)
+    } else if (val instanceof Array) {
+        let json = []
+        for (const item of val) {
+            json.push(toBase64(item))
+        }
+        return json;
+    } else if (typeof val == 'object') {
+        let json = {}
+        for (const key in val) {
+            if (Object.hasOwnProperty.call(val, key)) {
+                const element = val[key]
+                json[key] = toBase64(element)
+            }
+        }
+        return json;
+    }
+}
+
+function isJSONString(val) {
+    try {
+        let json = eval("(" + val + ")")
+        if (typeof json == 'object') {
+            return true;
+        }
+    } catch (e) {}
+    return false;
+}
+
+//check param is string and required
+function checkStringParam(param, required) {
+    if (param == '' || param == undefined) {
+        if (required) {
+            return false
+        } else {
+            return true
+        }
+    } else {
+        if (typeof (param) == 'string') {
+            return true
+        } else {
+            return false
+        }
+    }
+}
+
+const queryKeyId = (appid, keyid) => {
+    return {
+        selector: {
+            creator: appid,
+            keyid
+        },
+        fields: ['keyBlob'],
+        limit: 1
+    }
+}
+
+const queryDefaultCMK = (appid) => {
+    return {
+        selector: {
+            appid
+        },
+        fields: ['sm_default_cmk'],
+        limit: 1
+    }
+}
+
+function store_tags(DB, tags_jsonArray, appid, secretName) {
+    for (let i in tags_jsonArray) {
+        DB.insert({
+            _id: `secret_tags:${uuidv4()}`,
+            appid,
+            secretName,
+            tagKey: tags_jsonArray[i]['tagKey'],
+            tags_value: tags_jsonArray[i]['tagValue'],
+        })
+    }
+}
+
+//create Secret
+const createSecret = async (res, appid, payload, DB) => {
+    try {
+        const createTime = new Date().getTime()
+        let keyBlob = ''
+        let tags_jsonArray = ''
+        let secretData = getParam_String(payload, 'secretData')
+        let secretName = getParam_String(payload, 'secretName')
+        let versionId = getParam_String(payload, 'versionId')
+        let encryptionKeyId = getParam_String(payload, 'encryptionKeyId', false)
+        let description = getParam_String(payload, 'description')
+        let tags = getParam_String(payload, 'tags')
+        let rotationInterval = getParam_String(payload, 'rotationInterval')
+        if (!checkStringParam(secretData, true)) {
+            res.send(_result(400, 'secretData cannot be empty and must be string'))
+            return
+        }
+        if (!checkStringParam(secretName, true)) {
+            res.send(_result(400, 'secretName cannot be empty and must be string'))
+            return
+        }
+        if (!checkStringParam(versionId, true)) {
```

Comment:
versionId is a MUST string. 
if versionid is +_+, what's the next versionId after auto-rotation?